keras_core/utils/dataset_utils.py

```diff
@@ -6,44 +12,503 @@
 def split_dataset(
     dataset, left_size=None, right_size=None, shuffle=False, seed=None
 ):
-    """Splits a dataset into a left half and a right half (e.g. train / test).
+    """Split a dataset into a left half and a right half (e.g. train / test).
 
     Args:
-        dataset: A `tf.data.Dataset` object, or a list/tuple of arrays with the
-            same length.
+        dataset:
+            A `tf.data.Dataset or torchDataset`  object,
+            or a list/tuple of arrays with the
+          same length.
         left_size: If float (in the range `[0, 1]`), it signifies
-            the fraction of the data to pack in the left dataset.
-            If integer, it signifies the number of samples to pack
-            in the left dataset. If `None`, it defaults to the complement
-            to `right_size`.
+          the fraction of the data to pack in the left dataset. If integer, it
+          signifies the number of samples to pack in the left dataset. If
+          `None`, it uses the complement to `right_size`. Defaults to `None`.
         right_size: If float (in the range `[0, 1]`), it signifies
-            the fraction of the data to pack in the right dataset.
-            If integer, it signifies the number of samples to pack
-            in the right dataset. If `None`, it defaults to the complement
-            to `left_size`.
+          the fraction of the data to pack in the right dataset. If integer, it
+          signifies the number of samples to pack in the right dataset. If
+          `None`, it uses the complement to `left_size`. Defaults to `None`.
         shuffle: Boolean, whether to shuffle the data before splitting it.
         seed: A random seed for shuffling.
 
     Returns:
-        A tuple of two `tf.data.Dataset` objects: the left and right splits.
+        A tuple of two `tf.data.Dataset or torchDataset` objects:
+        the left and right splits.
 
     Example:
 
     >>> data = np.random.random(size=(1000, 4))
-    >>> left_ds, right_ds = split_dataset(data, left_size=0.8)
+    >>> left_ds, right_ds = keras_core.utils.split_dataset(data, left_size=0.8)
     >>> int(left_ds.cardinality())
     800
     >>> int(right_ds.cardinality())
     200
+
     """
-    # TODO: long-term, port implementation.
-    return tf.keras.utils.split_dataset(
-        dataset,
-        left_size=left_size,
-        right_size=right_size,
-        shuffle=shuffle,
-        seed=seed,
+    from torch.utils.data import Dataset as torchDataset
```

Comment:
- Do not import torch **unless** the dataset passed is a torch dataset.
- Just `import torch` then access `torch.utils.data.Dataset`.

To check whether it's a torch dataset, you can do something similar to this https://github.com/keras-team/keras-core/blob/main/keras_core/trainers/epoch_iterator.py#L228

Reply:
@fchollet nested import like` torch.utils.data.Dataset` is not working as module never gets added to globals hence throws import error this is same as I mentioned it [here](https://github.com/keras-team/keras-core/issues/463) and this still is reproducible.